### PR TITLE
Buildings honza dudek

### PR DESCRIPTION
**Pull request by HonzaDudek**

### CODE REVIEW

File: src/app/_models/building.model.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    } else if (type === 'factory') {
+      this.type = 'factory';
+    } else {
+      this.type = 'academy';
+    }
+    this.level = 1;
+    this.hp = 100;
+    this.started_at = Date.now();
+    this.finished_at = Date.now() + 1000;
+  }
```
Comment (adamgyulavari) [inline src/app/_models/building.model.ts:13]: @HonzaDudek have a general constructor here, and you should have BuildingFactory that calls the constructor with these parameters

File: src/app/_helpers/factories/building.factory.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import {ComponentFactory, ComponentFactoryResolver, FactoryProvider, OnInit} from '@angular/core';
+import { Building} from '../../_models/building.model';
+import {FactoryGenerator} from '@angular/compiler-cli/src/ngtsc/shims';
+
+export class BuildingFactor {
```
Comment (adamgyulavari) [inline src/app/_helpers/factories/building.factory.ts]: @HonzaDudek typo

File: src/app/_helpers/authentication/auth.guard.ts [hunk 3]
```diff
@@ -11,9 +11,7 @@ export class AuthGuard implements CanActivate {
               private router: Router) {
   }
 
-  canActivate(
-    next: ActivatedRouteSnapshot,
-    state: RouterStateSnapshot): boolean {
+  canActivate(): boolean {
```
Comment (adamgyulavari) [inline src/app/_helpers/authentication/auth.guard.ts:14]: @HonzaDudek only have changes in the pull request that are related to the feature

File: src/app/_helpers/interceptors/buildings.interceptor.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    return new Building(id, type);
+  }
+
+  findBuilding(buildingId: number): Building {
+    for (let i = 0; i < buildings.length; i++) {
+      if (buildings[i].id === buildingId) {
+        return buildings[i];
+      }
+    }
+  }
```
Comment (adamgyulavari) [inline src/app/_helpers/interceptors/buildings.interceptor.ts]: @HonzaDudek this should be extracted into a separate file with the example data

File: src/app/_helpers/interceptors/buildings.interceptor.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+          observer.complete();
+        });
+    } else if (req.url.endsWith(`/game/buildings/${req.body.id}`) && req.method === 'PUT') {
+      let buildingToUpdate = this.findBuilding(req.body.id);
+      buildingToUpdate.level += 1;
+      buildings[buildings.indexOf(this.findBuilding(req.body.id))] = buildingToUpdate;
+    } else if (req.url.endsWith('/game/buildings/' + req.body.id) && req.method === 'GET') {
+      let building = this.findBuilding(req.body.id);
+      return new Observable<any>(observer => {
+        observer.next(new HttpResponse<any>(
```
Comment (adamgyulavari) [inline src/app/_helpers/interceptors/buildings.interceptor.ts]: @HonzaDudek extract these into a separate method

File: src/app/game/buildings/building-details/building-detail.component.html [hunk 6]
```diff
@@ -0,0 +1,41 @@
+<div id="overlay">
+  <div id="modal-container">
+    <div class="building__details">
+      <img *ngIf="building.type === 'academy'" src="../../../../assets/images/buildings/academy.svg" alt="">
+      <img *ngIf="building.type === 'factory'" src="../../../../assets/images/buildings/factory.svg" alt="">
+      <img *ngIf="building.type === 'mine'" src="../../../../assets/images/buildings/mine.svg" alt="">
+      <img *ngIf="building.type === 'townhall'" src="../../../../assets/images/buildings/townhall.svg" alt="">
```
Comment (adamgyulavari) [inline src/app/game/buildings/building-details/building-detail.component.html]: ``

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

File: src/app/_helpers/interceptors/buildings.interceptor.ts (hunk 4 above)
Comment (HonzaDudek) [reply src/app/_helpers/interceptors/buildings.interceptor.ts]: Array and methods extracted to buildingsResponseMock

File: src/app/_helpers/factories/building.factory.ts (hunk 2 above)
Comment (HonzaDudek) [reply src/app/_helpers/factories/building.factory.ts]: Created building factory and corrected typo

File: src/app/_helpers/factories/building.factory.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import { Building} from '../../_models/building.model';
+
+export class BuildingFactory {
+
+  constructor() {
+  }
+
+  createBuildingComponent(id: number, type: string): Building {
```
Comment (adamgyulavari) [inline src/app/_helpers/factories/building.factory.ts]: @HonzaDudek it's not creating a component just a model, let's just call it `createBuilding`

File: src/app/_helpers/interceptors/login.interceptor.ts [hunk 8]
```diff
@@ -94,4 +71,19 @@ export class LoginInterceptor implements HttpInterceptor {
...
+          {
+            body: {
+              responseBody
+            },
+            status: status
+          }
+        ));
+        observer.complete();
+      });
+  }
```
Comment (adamgyulavari) [inline src/app/_helpers/interceptors/login.interceptor.ts]: @HonzaDudek well, if we're using it in multiple interceptors we should put it into some utilities/helper for the interceptors

File: src/app/game/buildings/building-details/building-detail.component.html [hunk 9]
```diff
@@ -0,0 +1,38 @@
+<div id="overlay" >
+  <div id="modal-container">
+    <div class="building__details">
+      <img [src]="imgSrc" alt="">
+      <table class="building__details--table">
+        <thead>
+        <tr >
+          <td colspan="2">{{building.type}}  details</td>
```
Comment (adamgyulavari) [inline src/app/game/buildings/building-details/building-detail.component.html:8]: @HonzaDudek two spaces on purpose?

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

File: src/app/game/buildings/building-details/building-detail.component.html (hunk 9 above)
Comment (HonzaDudek) [reply src/app/game/buildings/building-details/building-detail.component.html:8]: Yes, to center the text above both columns.

File: .vscode/launch.json [hunk 10]
```diff
@@ -0,0 +1,15 @@
+{
+  // Use IntelliSense to learn about possible attributes.
+  // Hover to view descriptions of existing attributes.
+  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
```
Comment (adamgyulavari) [inline .vscode/launch.json]: @HonzaDudek why is this here?

Comment (adamgyulavari) [reply .vscode/launch.json]: .vscode folder should be gitignored

File: src/app/_helpers/interceptors/resources.interceptor.ts [hunk 11]
```diff
@@ -10,33 +10,68 @@ import {
...
 
+// const resource: Resources[] = [
+//   {amount: 500,
+//     type: 'food',
+//     generation: 0 },
+
+//    {amount: 320,
+//     type: 'diamond',
+//     generation: 1 }
+// ];
```
Comment (adamgyulavari) [inline src/app/_helpers/interceptors/resources.interceptor.ts]: @HonzaDudek remove outcommented code

**Review by adamgyulavari — APPROVED** [review]
